modules/backup/variables.tf

```diff
@@ -1,6 +1,11 @@
 variable "resource_name_prefix" {
   description = "Resource name prefix used for tagging and naming AWS resources"
   type        = string
+
+  validation {
+    condition     = can(regex("^[a-zA-Z0-9-]+$", var.resource_name_prefix)) && !can(regex("^-", var.resource_name_prefix))
```

Comment:
This could be done with a single expression `^[a-zA-Z0-9-][a-zA-Z0-9-]+$`.
BTW, the resource name prefix is used in all modules, but only here the validation is changed.

Reply:
I have changed it to all the rest of the modules. About the expression - I have tested it and it works this way. If you want, I can test your expression as well and put it instead.
Thank you.